### ISSUE TITLE
How to use WeightedRandomSampler from PyTorch or write custom sampler

### ISSUE THREAD

**Issue by edsml-hmc122** [open-mmlab/mmsegmentation#3104]
Hi, I have an imbalanced dataset and I want to use PyTorch's `WeightedRandomSampler` or equivalent, so that each batch will be balanced during training. How can I accomplish this? Note this is not the same as using class weights to skew the loss, because with that technique, most batches would still contain only samples from the majority class.

For example (this code doesn't work):

```python
cfg.train_dataloader.sampler = dict(
    type='WeightedRandomSampler',
    shuffle=True,
    weights=sample_weights,  # 1 weight for each sample in the dataset
    num_samples=len(sample_weights),
    replacement=True  # True for oversampling
)
```

Since `WeightedRandomSampler` gives an error, is it possible to write my own Sampler and use it in the config?

PyTorch documentation for WeightedRandomSampler: https://pytorch.org/docs/stable/data.html#torch.utils.data.WeightedRandomSampler
Tutorial on how it's used (better explanation): https://www.youtube.com/watch?v=4JFVhJyTZ44

Cheers!
